### ISSUE TITLE
Multisite navigation is not visible after upgrading to craft 4 (and therefor navigation 2.0.4)

### ISSUE THREAD

**Issue by mrwijsman** [verbb/navigation#309]
### Describe the bug

After upgrading my multisite to craft 4 and navigation 2.0.4 there was not visible, however it was still in the database. 

I noticed the navigation_navs_site table was empty so  created a new navigation enabled for all sites which filled the table with content. 

Finally I changed the "navId" in the table "navigation_navs_site" for all sites from 2 (the new one) to 1(my old navigation) and voila, my old navigation reappeared.

### Steps to reproduce

1.  Upgrade your multisite to craft 4 and navigation 2.0.4

### Craft CMS version

Pro 4.2.2

### Plugin version

2.0.4

### Multi-site?

Yes

**Comment by engram-design** [comment]
If you re-save your navigations, does that populate the navigation for each site? It should be migrating them to new site settings we've implemented for Craft 4, but sounds like that might not be happening.

**Comment by mikewink** [comment]
I have a similar issue on two projects. Neither of them are multi-site installations. I did run `php craft resave/navigation-nodes`. It did not change the backend. The screen is empty. The navigation is still fully functional in the frontend. 

I'm running Craft CMS 4.2.3 with Navigation 2.0.4. This is straight after upgrading to Craft 4 from a Craft 3.7.x system. The update ran without any errors, so did the migrations. I cleared the caches and ran `php craft db/convert-charset` as recommended in the Craft CMS docs.

It has worked on a multi-site project that I have updated as well! There I can see the navigation for all my sites.

### Update

I had a look at the DB too and found that my two navigation entries, in one project, were both not `enabled` in `navigation_navs_sites`! I changed the value to `1`, and now it is working as expected again. 

The other project has no entries for `navigation_navs_sites` at all! So that trick does not work, and I might need to do the same as mrwijsman to fix it.

**Comment by engram-design** [comment]
Would either of you be willing to share your database for your Craft 3 install? I've been testing today migrating non-multi site navigations, but unable to reproduce this behaviour I'm afraid. Sending them to `support at verbb.io` would be amazing if possible.

**Comment by mikewink** [comment]
Hello Josh,

I'm upgrading several more sites over the weekend. I did one more after writing the first comment and update on it. It is a single site too, and here the process worked as expected.

I'll continue to migrate three more projects and will let you know my findings. Let me talk to one of the project leads and if she agrees I'll give you some access.

**Comment by DanielDarrenJones** [comment]
I was also hitting this issue, @mikewink seems to be correct, the entries in `navigation_navs_sites` were not enabled by default after migrating (I also just updated from craft 3).

Updating this to be enabled appears to have solved the problem.

**Comment by engram-design** [comment]
If anyone can provide their Craft 3 database, or at the very least a copy of their `navigation_navs` table before the upgrade (when still in Craft 3), that would be great. I can't replicate this issue on several upgrades (single and multi site).

As for the change, we introduced showing only site-enabled navigations due to https://github.com/verbb/navigation/issues/308 in the navigation index. But this really is probably just masking an underlying issue where navigations might not have been added to `navigation_navs_sites` in the first place, which they should.

**Comment by engram-design** [comment]
Actually, managed to figure this out. Fixed in [2.0.5](https://github.com/verbb/navigation/releases/tag/2.0.5).

A migration should also fire to fix up anyone on a single-site install with this issue. If you're finding there's still an issue, you can manually edit your `navigation_navs_sites` database table to enable all your navigations are enabled.

**Comment by mikewink** [comment]
Your fix did not work on a fresh upgrade. The table was empty after the migrations ran. I'm on Craft CMS 4.2.3 and navigation 2.0.5. PHP is version 8.1.9 and MySQL 5.7.36. 

I'll handle the update manually.

**Comment by engram-design** [comment]
@mikewink Is your site a multi-site? I could only replicate this for a single site install. Can you share your `navigation_*` tables (before the upgrade)?

**Comment by mikewink** [comment]
It is a single site and here is the link to the dumps, before and after.

https://nx15846.your-storageshare.de/s/cYgFr3owL4zoE7X

**Comment by engram-design** [comment]
So those navigations might not've been touched in your Craft 3 install for a while, as they have different settings that we're not catering for.

Should be fixed for the next release. To get the fix early, change your `verbb/navigation` requirement in `composer.json` to:

```
"require": {
  "verbb/navigation": "dev-craft-4 as 2.0.5",
  "...": "..."
}
```

Then run `composer update`.

**Comment by mikewink** [comment]
@engram-design Some of those projects run in production for about two years or so. I did all the upgrades and migrations. 🤔 

Thank you very much for the update and fix! 👍🏼 Will test it out on Monday. 🏢

**Comment by engram-design** [comment]
Fixed in [2.0.6](https://github.com/verbb/navigation/releases/tag/2.0.6)
